From 7ec502b5f65b6db10d628f4b6bcbf05f3b79a274 Mon Sep 17 00:00:00 2001
From: Ram Chandra Jangir <rjangir@codeaurora.org>
Date: Thu, 4 Mar 2021 15:42:16 +0530
Subject: [PATCH] include: Add support to use user configured linux config file

Currently OpenWrt uses linux config name as config-x.y, which
means we can have a single config-x.y for a linux-x.y.

We will add a CONFIG_EXTERNAL_LINUX_CONFIG_NAME, which gives
freedom to the users for keeping multiple config files for
same linux version. OpenWrt framework will read config name
from CONFIG_EXTERNAL_LINUX_CONFIG_NAME and will use it for
building the linux.

Change-Id: I8a92392232ed419d97af2b5fbe67fa61a186d2c9
Signed-off-by: Ram Chandra Jangir <rjangir@codeaurora.org>
(cherry picked from commit 007654a42f06d955433090211332e1bd36aa4472)
(cherry picked from commit 3e67c354573fd9a37161b5729d390a4215c0d55d)
(cherry picked from commit 2c1cb56b81e2d9d5f1662ba6205d9cccd0924360)
---
 include/target.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/target.mk b/include/target.mk
index 386abf2ef1e2..9b6f649b29d2 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -177,11 +177,15 @@ GENERIC_PATCH_DIR := $(GENERIC_PLATFORM_DIR)/pending$(if $(wildcard $(GENERIC_PL
 GENERIC_HACK_DIR := $(GENERIC_PLATFORM_DIR)/hack$(if $(wildcard $(GENERIC_PLATFORM_DIR)/hack-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
 GENERIC_FILES_DIR := $(foreach dir,$(wildcard $(GENERIC_PLATFORM_DIR)/files $(GENERIC_PLATFORM_DIR)/files-$(KERNEL_PATCHVER)),"$(dir)")
 
-__config_name_list = $(1)/config-$(KERNEL_PATCHVER) $(1)/config-default
+EXTERNAL_LINUX_CONFIG = $(call qstrip,$(CONFIG_EXTERNAL_LINUX_CONFIG_NAME))
+__config_name_list = $(if $(EXTERNAL_LINUX_CONFIG),$(1)/$(EXTERNAL_LINUX_CONFIG),$(1)/config-$(KERNEL_PATCHVER)) $(1)/config-default
 __config_list = $(firstword $(wildcard $(call __config_name_list,$(1))))
 find_kernel_config=$(if $(__config_list),$(__config_list),$(lastword $(__config_name_list)))
 
 GENERIC_LINUX_CONFIG = $(call find_kernel_config,$(GENERIC_PLATFORM_DIR))
+ifeq ($(wildcard $(GENERIC_LINUX_CONFIG)),)
+GENERIC_LINUX_CONFIG = $(GENERIC_PLATFORM_DIR)/config-$(KERNEL_PATCHVER)
+endif
 LINUX_TARGET_CONFIG = $(call find_kernel_config,$(PLATFORM_DIR))
 ifneq ($(PLATFORM_DIR),$(PLATFORM_SUBDIR))
   LINUX_SUBTARGET_CONFIG = $(call find_kernel_config,$(PLATFORM_SUBDIR))
-- 
2.34.1

